ui/pages/Dashboard/components/ProjectsTable.jsx

```diff
@@ -163,7 +175,7 @@ const ProjectsTable = React.memo(({ visibleProjects, loading, load, user }) =>
         emptyContent="0 projects found"
         loading={loading}
         data={visibleProjects}
-        columns={user.isStaff ? STAFF_COLUMNS : COLUMNS}
+        columns={COLUMNS.slice(0, googleLoginEnabled && user.isAnvil ? 2 : 1).concat(user.isStaff ? STAFF_COLUMNS.slice(2) : COLUMNS.slice(2))}
```

Comment:
you never define a list in props definitions in react because it causes unnecessary re-renders, and `COLUMNS.slice(` and `concat` create new lists. You unfortunately need to define 4 top-level `COLUMNS` lists for all the possible permutations of staff and google login, and then probably make a helper function to return the correct one

Reply:
There is a no-nested-ternary rule so that I can't select one from these 4 columns. `.slice()` and `.concat()` create new objects. That's not good. How can the helper function not to create new objects?

Reply:
```
COLUMNS = ...
STAFF_COLUMNS = ...
ANVIL_COLUMNS = COLUMNS.splice(...)
STAFF_ANVIL_COLUMNS = COLUMNS.splice(...)

const getColumns = (isAnVIl, isStaff) => {
  if (isAnvil) {
    return isStaff? STAFF_ANVIL_COLUMNS : ANVIL_COLUMNS
  } else { 
      return isStaff? STAFF_COLUMNS : COLUMNS
  }
 }
}
```

Reply:
I have done it similarly.